### PR TITLE
Allow custom ObjectIds when uploading to GridFS

### PR DESCRIPTION
**Pull request by rozza**
JAVA-1983

### CODE REVIEW

File: driver-core/src/main/com/mongodb/client/gridfs/model/GridFSUploadOptions.java [hunk 1]
```diff
@@ -77,4 +80,25 @@ public GridFSUploadOptions metadata(final Document metadata) {
...
+        return fileId;
+    }
+
+    /**
+     * Sets fileId for the uploaded file
+     *
+     * @param fileId the fileId to use for the uploaded file
+     * @return this
+     */
+    public GridFSUploadOptions fileId(final ObjectId fileId) {
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/client/gridfs/model/GridFSUploadOptions.java]: Keep the constructors together before all other methods

Comment (rozza) [inline driver-core/src/main/com/mongodb/client/gridfs/model/GridFSUploadOptions.java]: ? Thats not a constructor - just part the builder

Comment (jyemin) [inline driver-core/src/main/com/mongodb/client/gridfs/model/GridFSUploadOptions.java]: Need coffee.... Acknowledged :(

**Comment by rozza** [comment]
Closing as this is a spec breaking change

**Pull request closed without merging**